contributors/design-proposals/scheduling/scheduling-policy.md

```diff
@@ -0,0 +1,781 @@
+    affinities:
+      nodeAffinities:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/region","kubernetes.io/authorized-region"]
+              operators: ["In","NotIn"]
+              values: ["eu-2", "us-1"]
+      podAntiAffinities: {}
+  default:
+    affinity:
+      nodeAffinity:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["beta.kubernetes.io/arch"]
+              operators: ["In"]
+              values: ["amd64"]
+```
+
+In this example, we allow:
+- hard NodeAffinity based on
+  -  `beta.kubernetes.io/arch` if value is `amd64` or `arm64`. Defaults to `amd64` if unspecified.
+  -  (optionally) any combination of <keys, operators, values> specified in `allowed` secion.
+- All podAntiAffinities
+- No podAffinities
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities:
+      nodeAffinities:
+        requiredDuringSchedulingIgnoredDuringExecution:
+          nodeSelectorTerms:
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/region","kubernetes.io/authorized-region"]
+              operators: ["In","NotIn"]
+              values: ["eu-2", "us-1"]
+          - matchExpressions:
+            - keys: ["failure-domain.beta.kubernetes.io/zone"]
+              operators: ["NotIn"]
+              values: ["dc1", "dc2"]
+          podAntiAffinities: {}
+```
+
+This example highlights the case where you don't want a full combinatory, here we allow the same affinities as the previous example, in addition,
+we allow a match expression that checks that some zone labels are not in the specified values.
+
+
+##### Allow-all policy
+In this example, all affinities are allowed:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities: {}
+```
+
+Which is equivalent to:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowed:
+    affinities:
+      nodeAffinities: {}
+      podAffinities: {}
+      podAntiAffinities: {}
+```
+
+If a sub-item of allowedAffinities is absent from SchedulingPolicy, it is not allowed e.g:
+
+```yaml
+Kind: SchedulingPolicy
+spec:
+  allowedAffinities:
+    nodeAffinities: {}
+```
+
+In this example, only nodeAffinities (required and preferred) are allowed but no podAffinities nor podAntiAffinities.
+
+## Multiple SchedulingPolicies considerations
+
+NOTE: here a merge is the set of resulting authorizations after going through the available policies (i.e. we do not aggregate policies into a newly created `SchedulingPolicy`)
+
+several merging strategies are being considered.
+
+### Option1: smart deep merge
+
+If RBAC permissions provide a serviceaccount a way to use several schedpols, conflict resolution must occur. The proposed behaviour is:
+
+-  `required` fields use an alphabetic order and then a first-seen-wins strategy for each sub-keys
```

Comment:
This seems too easy to accidentally open something up by triggering a conflict. I'd suggest just taking the intersection of the policies, i.e. a Pod must meet the requirements of both. If there is a conflict, just deny fail closed and deny the pod.